test/unittest/tests/test_assertions.cpp

```diff
@@ -0,0 +1,48 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "utils.h"
+#include "wiredtiger.h"
+#include "wrappers/connection_wrapper.h"
+#include "wt_internal.h"
+
+/* Assert that a WT assertion has fired with the expected message, then clear the flag and message. */
+void
+expect_assertion(WT_SESSION_IMPL *session, std::string expected_message)
+{
+    REQUIRE(session->unittest_assert_hit);
+    REQUIRE(std::string(session->unittest_assert_msg) == expected_message);
+
+    // Clear the assertion flag and message for the next test step
+    session->unittest_assert_hit = false;
+    memset(session->unittest_assert_msg, 0, WT_SESSION_UNITTEST_BUF_LEN);
+}
+
+void
+expect_no_assertion(WT_SESSION_IMPL *session)
+{
+    REQUIRE_FALSE(session->unittest_assert_hit);
+    REQUIRE(std::string(session->unittest_assert_msg).empty());
+}
+
+TEST_CASE("Simple implementation of unit testing WT_ASSERT_ALWAYS", "[assertions]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
```

Comment:
Can I suggest adding an additional call to `expect_no_assertion(session)` after the line `WT_SESSION_IMPL *session = conn.createSession()` to double check that the session's status was set up as expected.

Reply:
Good point! Added in [4076e6c](https://github.com/wiredtiger/wiredtiger/pull/8474/commits/4076e6c78e2aa72e641af026b35836e81b8120fc)